pkg/orchestrator/connectors.go

```diff
@@ -163,3 +165,42 @@ func (c *ConnectorOrchestrator) Update(ctx context.Context, id string, config co
 	r.Skip()
 	return conn, nil
 }
+
+func (c *ConnectorOrchestrator) Validate(
+	ctx context.Context,
+	t connector.Type,
+	config connector.Config,
+) error {
+	logger := log.CtxLogger{Logger: zerolog.Nop()}
+	d, err := c.plugins.NewDispenser(logger, config.Plugin)
+	if err != nil {
+		return cerrors.Errorf("couldn't get dispenser: %w", err)
+	}
+
+	switch t {
+	case connector.TypeSource:
+		src, _ := d.DispenseSource()
+		err := src.Configure(ctx, config.Settings)
+		if err != nil {
+			return cerrors.Errorf("configurations failed: %w", err)
+		}
+		err = src.Teardown(ctx)
+		if err != nil {
+			return cerrors.Errorf("couldn't teardown the connector: %w", err)
+		}
```

Comment:
I'm wondering, does a client care if the teardown didn't succeed? It's more like a cleanup type of thing, which we can turn a blind eye to.